src/cpr_data_access/utils.py

```diff
@@ -0,0 +1,67 @@
+import csv
+from pathlib import Path
+
+
+def is_sensitive_query(text: str, sensitive_terms: set) -> bool:
+    """
+    Scans text to determine if the query should be considered sensitive
+
+    It does this by evaluating it against a set of predefined sensitive terms.
+    These, as well as the specific logic are reproduced from previous work, which
+    stated that "If the query contains any sensitive terms, and the length of the
+    shortest sensitive term is >=50% of the length of the query by number of words..."
+    then it is considered sensitive. Further details on the original can be found here:
+    https://github.com/climatepolicyradar/navigator/pull/815
+
+    This updated version builds on the above to avoid a loophole where a string of
+    sensitive terms can be incorrectly flagged as not being sensitive. This happens
+    because it is the shortest sensitive term that is compared to the rest of the
+    query, and the rest of the query at that point can contain other sensitive terms.
+
+    """
+    sensitive_terms_in_query = [
```

Comment:
`set(map(str.lower, sensitive_terms)).intersection(set(text.lower().split()))`?

Reply:
I'm not sure we can use `text.lower().split()` here because some of the sensitive terms are multiple words

Reply:
Just profiled both options and seems set intersection is a lot slower as well, unfortunately